### PR TITLE
Gists

### PR DESCRIPTION
**Pull request by alexandreneves**

### CODE REVIEW

File: src/domain/documentStoreState/documentStoreStateSelectors.ts [hunk 1]
```diff
@@ -37,7 +40,6 @@ export function createNewPlantUml(): Document {
     id,
     name: `New Diagram` as DocumentName,
     code: `New->Diagram` as DocumentContent,
-    type: DocumentContentType.PLANT_UML
```
Comment (nevcos) [inline src/domain/documentStoreState/documentStoreStateSelectors.ts]: Should be kept, no?

File: src/remoteApi/githubApi.ts [hunk 2]
```diff
@@ -12,25 +14,19 @@ export async function getToken(code: string): Promise<any> {
...
+export async function getGists(): Promise<Gist[]> {
   const token = Cookies.get(GITHUB_TOKEN_COOKIE_KEY);
-  const data = await request(GITHUB_ENDPOINT + "/gists", {
+  const data = (await request(GITHUB_ENDPOINT + "/gists", {
     headers: {
       Accept: "application/vnd.github.v3+json",
       Authorization: `token ${token}`
     }
-  });
+  })) as unknown as Gist[];
```
Comment (nevcos) [inline src/remoteApi/githubApi.ts]: `as unknown` 😈

File: src/service/gitHub.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+export async function getFilesFromGists(gists: Gist[]): Promise<DocumentMap> {
+  const files: DocumentMap = {};
+
+  await Promise.all(
+    gists.map(async (gist) => {
+      const folderId = gist.id as FolderId;
+
+      await Promise.all(
+        Object.keys(gist.files).map(async (key) => {
+          const file = gist.files[key];
```
Comment (nevcos) [inline src/service/gitHub.ts:34]: Only as a FYI, I think this works as well, but needs confirmation:
```suggestion
        Object.values(gist.files).map(async ([key, file]) => {
```

File: src/ui/sidebar/Sidebar.style.ts [hunk 4]
```diff
@@ -1,5 +1,5 @@
 import styled, { css } from "styled-components";
-import { Link } from 'react-router-dom';
+import { Link } from "react-router-dom";
```
Comment (nevcos) [inline src/ui/sidebar/Sidebar.style.ts:2]: 😊

File: src/util/file.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import { DocumentContentType } from "../domain/document/DocumentContentType";
+
+export function getFileTypeFromExtension(filename: string): DocumentContentType {
```
Comment (nevcos) [inline src/util/file.ts]: Might make sense move this to `domain/document` given it's business logic.

**Review by nevcos — APPROVED** [review]
Outstanding work 🚀 
Left just some minor suggestions

File: src/domain/documentStoreState/documentStoreStateSelectors.ts (hunk 1 above)
Comment (alexandreneves) [reply src/domain/documentStoreState/documentStoreStateSelectors.ts]: Correct!

File: src/ui/sidebar/Sidebar.style.ts (hunk 4 above)
Comment (alexandreneves) [reply src/ui/sidebar/Sidebar.style.ts:2]: :D

**Pull request merged**